### PR TITLE
Allow to connect to custom infra

### PR DESCRIPTION
**Pull request by jakubno**

### CODE REVIEW

File: packages/python-sdk/e2b/sandbox/sandbox_connection.py [hunk 1]
```diff
@@ -245,16 +258,16 @@ def _open(self, timeout: Optional[float] = TIMEOUT) -> None:
             self._connect_rpc(timeout)
         except Exception as e:
             print(e)
+            print(self.id)
+            sleep(10000)
```
Comment (jakubno) [inline packages/python-sdk/e2b/sandbox/sandbox_connection.py]: Remove

File: packages/python-sdk/e2b/sandbox/sandbox_connection.py [hunk 2]
```diff
@@ -245,16 +258,16 @@ def _open(self, timeout: Optional[float] = TIMEOUT) -> None:
...
             self._close()
             raise e
 
     def _connect_rpc(self, timeout: Optional[float] = TIMEOUT):
-        hostname = self.get_hostname(self._debug_port or ENVD_PORT)
-        protocol = "ws" if self._debug_dev_env == "local" else "wss"
-
-        sandbox_url = f"{protocol}://{hostname}{WS_ROUTE}"
+        sandbox_url = self.get_sandbox_url(self._debug_port or ENVD_PORT)
```
Comment (jakubno) [inline packages/python-sdk/e2b/sandbox/sandbox_connection.py]: fix this, you have to use `ws`

**Comment by ValentaTomas** [comment]
Not sure if we want to replace getHostname/get_hostname

**Comment by jakubno** [comment]
> Not sure if we want to replace getHostname/get_hostname

What are you concerns exactly? I just wrapped it inside the getSandboxUrl to not repeat the code multiple times

**Pull request merged**